### PR TITLE
docs: update public symbol and module documentation style

### PR DESCRIPTION
**Pull request by iuioiua**
To accompany #4569

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -64,31 +64,35 @@ plain language, as follows:
...
-1. If it is a function or method, a description of the return value using the
-   [`@returns`](https://jsdoc.app/tags-returns) tag.
-1. At least one minimal example code snippet using the
-   [`@example`](https://jsdoc.app/tags-example) tag for basic usage. The reader
-   should be able to copy and execute the code snippet with the expected values.
+1. A description, first
+1. [`@param`](https://jsdoc.app/tags-param) tags for each parameter and a
+   [`@returns`](https://jsdoc.app/tags-returns) tag, if the symbol is a
+   function.
+1. A least one example code snippet using the
```
Comment (kt3k) [inline .github/CONTRIBUTING.md:79]: typo?
```suggestion
1. At least one example code snippet using the
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -64,31 +64,35 @@ plain language, as follows:
...
+   [`@example`](https://jsdoc.app/tags-example) tag and a title. The code
+   reproducible when copied and pasted as a script.
```
Comment (kt3k) [inline .github/CONTRIBUTING.md:81]:
```suggestion
   is reproducible when copied and pasted as a script.
```

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -64,31 +64,35 @@ plain language, as follows:
...
-  and [documentation](https://jsr.io/@std/datetime/doc/~/weekOfYear))
+Module files, or `mod.ts` files, should have the following:
+
+1. A high-level description of the package.
+1. Sections providing brief overviews of the APIs within the package, including
+   minimal example code snippets (without the `@example` tag).
+1. A [`@module`](https://jsdoc.app/tags-module) to denote module documentation.
+
+See the source code for
+[`std/datetime/mod.ts`](https://github.com/denoland/deno_std/blob/main/datetime/mod.ts)
```
Comment (kt3k) [inline .github/CONTRIBUTING.md:97]: Maybe better to link to one of the stable modules? (`datetime` can be changed drastically (could be removed) when Temporal becomes stable

Comment (iuioiua) [reply .github/CONTRIBUTING.md:97]: `std/datetime` is the only package with the correct documentation style now. I'll likely update this later.

**Pull request merged**